pkg/util/test/util.go

```diff
@@ -643,6 +643,74 @@ func validateImageTag(tag, namespace string, listOptions map[string]string) erro
 	return nil
 }
 
+func validateMonitoring(cluster *corev1alpha1.StorageCluster, timeout, interval time.Duration) error {
+	if cluster.Spec.Monitoring != nil && *cluster.Spec.Monitoring.EnableMetrics {
```

Comment:
On enable metrics/export metrics we only create a service monitor and prometheus rule, assuming this is needed for an external prometheus.
To deploy prometheus, `*cluster.Spec.Monitoring.Prometheus.Enabled` has to be set. So we should check the prometheus operator deployment (px-prometheus-operator) and prometheus deployment (px-prometheus) only when the prometheus.Enabled flag is set.